mods/d2k/maps/harkonnen-07/harkonnen07.lua

```diff
@@ -0,0 +1,238 @@
+		player.MarkFailedObjective(CaptureAtreidesConYard)
+	end
+
+	if (AHiTechFactory.IsDead or AHiTechFactory.Owner ~= atreides_main) and not HiTechIsDead then
+		Media.DisplayMessage("High Tech Factory neutralized! Atreides cut off from Imperial reinforcement!", "Mentat")
+		HiTechIsDead = true
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(10) == 0 and LastHarvesterEaten[ordos_main] then
+		local units = ordos_main.GetActorsByType("harvester")
+
+		if #units > 0 then
+			LastHarvesterEaten[ordos_main] = false
+			ProtectHarvester(units[1], ordos_main, AttackGroupSize[Difficulty])
+		end
+	end
+
+	if DateTime.GameTime % DateTime.Seconds(10) == 0 and LastHarvesterEaten[ordos_small] then
+		local units = ordos_small.GetActorsByType("harvester")
+
+		if #units > 0 then
+			LastHarvesterEaten[ordos_small] = false
+			ProtectHarvester(units[1], ordos_small, AttackGroupSize[Difficulty])
+		end
+	end
+end
+
+WorldLoaded = function()
+	atreides_main = Player.GetPlayer("Atreides Main Base")
+	atreides_small = Player.GetPlayer("Atreides Small Base")
+	corrino = Player.GetPlayer("Corrino")
+	player = Player.GetPlayer("Harkonnen")
+
+	InitObjectives(player)
+	CaptureAtreidesConYard = player.AddPrimaryObjective("Capture the Atreides Construction Yard.")
+	KillAtreides = player.AddPrimaryObjective("Destroy the Atreides.")
+	KillCorrino = player.AddPrimaryObjective("Destroy the Corrino.")
+	KillHarkonnen1 = atreides_main.AddPrimaryObjective("Kill all Harkonnen units.")
+	KillHarkonnen2 = atreides_small.AddPrimaryObjective("Kill all Harkonnen units.")
+	KillHarkonnen3 = corrino.AddPrimaryObjective("Kill all Harkonnen units.")
+
+	Camera.Position = HEngineer.CenterPosition
+	AtreidesAttackLocation = AConYard2.Location
+
+	Trigger.AfterDelay(DateTime.Minutes(5), function()
+		SendAirStrike()
+	end)
+
+	Trigger.OnAllKilledOrCaptured(AtreidesMainBase, function()
+		Utils.Do(atreides_main.GetGroundAttackers(), IdleHunt)
+	end)
+
+	Trigger.OnAllKilledOrCaptured(AtreidesSmallBase, function()
+		Utils.Do(atreides_small.GetGroundAttackers(), IdleHunt)
+	end)
+
+	Trigger.OnAllKilledOrCaptured(CorrinoBase, function()
+		Utils.Do(corrino.GetGroundAttackers(), IdleHunt)
+	end)
+
+	local atreidesWaveCondition = function() return player.IsObjectiveCompleted(KillAtreides) or AHiTechFactory.IsDead or AHiTechFactory.Owner ~= atreides_main end
+	local corrinoWaveCondition = function() return player.IsObjectiveCompleted(KillCorrino) or AHiTechFactory.IsDead or AHiTechFactory.Owner ~= atreides_main end
+	local huntFunction = function(unit)
+		unit.AttackMove(AtreidesAttackLocation)
+		IdleHunt(unit)
+	end
+	SendEnemyReinforcements(atreides_main, EnemyAttackDelay[Difficulty], AtreidesPath, AtreidesReinforcements[Difficulty], atreidesWaveCondition, huntFunction)
+	SendEnemyReinforcements(corrino, EnemyAttackDelay[Difficulty], CorrinoPaths[1], CorrinoReinforcements[Difficulty][1], corrinoWaveCondition, huntFunction)
```

Comment:
Using `EnemyAttackDelay[Difficulty]` for all 3 makes all of them attack at the same time. I'd prefer if at least at 'easy', possibly at normal too, Corrino would always attack some 30-45 seconds after the Atreides, so less talented players (like me) don't have to deal with 3 attacks at once.